### ISSUE TITLE
Minio Server file download empty

### ISSUE THREAD

**Issue by harrycarp** [minio/minio#14071]
Whenever a file is downloaded from Minio, it downloads an empty file titled "download".
The same happens when previewing a file (if the file is an image, it shows a "broken" image icon).

## Expected Behavior
The file should be downloaded with content and the correct filename & extension.
When previewing the file, it should display the appropriate file and not attempt to download it.

## Current Behavior
When downloading a file, it downloads an empty file with the name "download" and no extension.
When previewing a file, if the file is a picture, it will display an empty "broken image" icon; if it is not a picture, it will download an empty "download" file with no extension.

![minio_bug_file_document](https://user-images.githubusercontent.com/33830630/148767923-d92d3994-c75a-4686-8ed5-7ee56c6d3b11.png)
![minio_bug_file_image](https://user-images.githubusercontent.com/33830630/148767940-a59e7954-ab4d-4249-9474-901b0c0b78a1.png)

This does not appear to be the case when running Minio Server in a docker container on MacOS (M1 ARM, MacBook Air). 
It is happening on Docker with a host running Ubuntu 21.04 LTS.

## Steps to Reproduce (for bugs)

1. Start the Minio Server docker container on an Ubuntu host
2. Create a bucket and upload a file to it (through the web UI or otherwise)
3. Attempt to preview or download the file

NOTE: Still yet to properly test running on a Docker Windows host and another Docker Linux host.

## Context
Need to view files and possibly download them through the Minio Server UI, whilst it is not mission critical as our applications use the Minio SDK to upload/fetch files, it would be good to be able to properly download and view the files.

## Regression
No/Unsure as it works on MacOS, also through a docker container.

## Your Environment

* Version: RELEASE.2022-01-08T03-11-54Z
* Operating System: Ubuntu 20.04
* Running on: Docker
* Image: minio/minio:latest (as of 2022-01-10)

**Comment by harshavardhana** [comment]
No idea here does your object have proper content-type set ?

**Comment by klauspost** [comment]
@harrycarp I cannot reproduce. Tested on `play.minio.io` with Chrome+Windows browser.

**Comment by harshavardhana** [comment]
@dvaldivia do you know if there is an issue here?

**Comment by harshavardhana** [comment]
> It is happening on Docker with a host running Ubuntu 21.04 LTS.

what is the filesystem here?

**Comment by harrycarp** [comment]
> > It is happening on Docker with a host running Ubuntu 21.04 LTS.
> 
> what is the filesystem here?

@harshavardhana The filesystem is `ext4`, when uploading a `text/plain` file-type, the Content-Type is displayed in the Minio UI under the Object Metadata, however when uploading any other type, it is not shown. 
<img width="1151" alt="Screen Shot 2022-01-11 at 8 38 44 am" src="https://user-images.githubusercontent.com/33830630/148861325-4dddd5b6-3bb0-4e29-a8d5-7637c8218514.png">
<img width="712" alt="Screen Shot 2022-01-11 at 8 38 32 am" src="https://user-images.githubusercontent.com/33830630/148861336-660da5fe-fabb-406c-ac4a-c35c1438124d.png">

Viewing the file through a public link; and download through the SDK (JavaScript) both work, and return the correct file.

**Comment by harrycarp** [comment]
Issue has been found to be the server hosting the Minio container, as the Minio container functions properly on another server running an identical specification. I will continue to look into this and update any findings in case anybody else has the same issue.

Thanks all for your help!

**Comment by harrycarp** [comment]
I just discovered the issue and it was not to do with the server it was hosting on, but instead to do with the Server URL behind a reverse proxy (NGINX).

When configuring the **MINIO_SERVER_URL** to point to a domain behind a reverse proxy to the container, it failed to get the file preview through the UI. 

When I **removed** the **MINIO_SERVER_URL** environment variable, so it auto set to the docker container IP with port 9000, the fetching of the file preview and downloading of the file through the browser worked.

Does anybody know why this might be the case? The MINIO_SERVER_URL listed is valid, and files can be uploaded/downloaded through the URL using the SDK.

**Comment by harshavardhana** [comment]
> Does anybody know why this might be the case? The MINIO_SERVER_URL listed is valid, and files can be uploaded/downloaded through the URL using the SDK.

well if you set this to be a wrong value @harrycarp then it won't be able to download things properly - can you show us what you set?

**Comment by harshavardhana** [comment]
> When I **removed** the **MINIO_SERVER_URL** environment variable, so it auto set to the docker container IP with port 9000, the fetching of the file preview and downloading of the file through the browser worked.

If you change this URL - this URL needs to be accessible and resolvable for the console to keep things working. If you do set it to an incorrect value it will not work.

We have documented this here which explains how to run `nginx` LB across console and MinIO ports you can of course have any server_name that you want here for your local setup. If you do have specific requirements make sure that 

MINIO_SERVER_URL points to lets say a TLS terminated LB then it needs to be publicly resolvable. 

I think this is just a pure setup issue here. Closing

**Comment by harshavardhana** [comment]
Here is our https://play.minio.io:9443 configuration

```
MINIO_VOLUMES="/home/play/data{1...4}"
MINIO_UPDATE=off
MINIO_SITE_REGION="us-east-1"
MINIO_ROOT_USER="Q3AM3UQ867SPQQA43P2F"
MINIO_ROOT_PASSWORD="zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG"
MINIO_KMS_KES_ENDPOINT="https://play.min.io:7373"
MINIO_KMS_KES_KEY_FILE="/home/play/root.key"
MINIO_KMS_KES_CERT_FILE="/home/play/root.cert"
MINIO_KMS_KES_KEY_NAME=my-minio-key
MINIO_PROMETHEUS_AUTH_TYPE="public"
MINIO_DELETE_CLEANUP_INTERVAL=5s
MINIO_SCANNER_CYCLE=30s
MINIO_PROMETHEUS_URL="http://localhost:9091"
MINIO_OPTS="--address play.min.io:9000 --console-address play.min.io:9443"
MINIO_DOMAIN="play.min.io,play.minio.io"
MINIO_API_REQUESTS_MAX=1000
```

**Comment by brknapik** [comment]
Hi 🙂 ,

I would like to come back to this topic regarding changes after updating from `RELEASE.2021-12-27T07-23-18Z` to `RELEASE.2022-04-12T06-55-35Z`.

We found the same issue regarding to download files as a empty file. From GUI we have all information about specified file like:
![Clipboard - 6 maja 2022 10_12](https://user-images.githubusercontent.com/93321802/167094530-c95c6e3a-05b7-450a-9a4e-8f999e4fcccc.png)

Is security bugfix could change this possibility ? I found workaround for that, but I don't know that bug, could be issue in our configuration (docker-compose). In our case, the solution is download via copying share path file, and after that we can get specified file.

We are using specified env variables in our docker compose file:

```
MINIO_ROOT_USER: {{ minio_root_usr }}
MINIO_ROOT_PASSWORD: {{ minio_root_pass }}
MINIO_PROMETHEUS_AUTH_TYPE: public
MINIO_PROMETHEUS_URL: {{prometheus_url}}
MINIO_BROWSER_REDIRECT_URL: {{ minio_console_url }}
MINIO_SERVER_URL:  {{ minio_server_url }}
MINIO_SCANNER_CYCLE: 30s
MINIO_UPDATE: "off"
```


Thanks for any advice
